addons/adyen_platforms/models/adyen_bank_account.py

```diff
@@ -0,0 +1,152 @@
+
+
+class AdyenBankAccount(models.Model):
+    _name = 'adyen.bank.account'
+    _description = 'Adyen for Platforms Bank Account'
+
+    adyen_account_id = fields.Many2one('adyen.account', ondelete='cascade')
+    bank_account_reference = fields.Char('Reference', default=lambda self: uuid.uuid4().hex)
+    bank_account_uuid = fields.Char('UUID')  # Given by Adyen
+    owner_name = fields.Char('Owner Name', required=True)
+    country_id = fields.Many2one('res.country', string='Country', domain=[('code', 'in', ADYEN_AVAILABLE_COUNTRIES)], required=True)
+    country_code = fields.Char(related='country_id.code')
+    currency_id = fields.Many2one('res.currency', string='Currency', required=True)
+    currency_name = fields.Char(related='currency_id.name', string='Currency Name')
+    iban = fields.Char('IBAN')
+    account_number = fields.Char('Account Number')
+    branch_code = fields.Char('Branch Code')
+    bank_city = fields.Char('Bank City')
+    bank_code = fields.Char('Bank Code')
+    bank_name = fields.Char('Bank Name')
+    account_type = fields.Selection(string='Account Type', selection=[
+        ('checking', 'Checking'),
+        ('savings', 'Savings'),
+    ])
+    owner_country_id = fields.Many2one('res.country', string='Owner Country')
+    owner_state_id = fields.Many2one('res.country.state', 'Owner State', domain="[('country_id', '=?', owner_country_id)]")
+    owner_street = fields.Char('Owner Street')
+    owner_city = fields.Char('Owner City')
+    owner_zip = fields.Char('Owner ZIP')
+    owner_house_number_or_name = fields.Char('Owner House Number or Name')
+
+    bank_statement = fields.Binary('Bank Statement', help="You need to provide a bank statement to allow payouts. \
+        The file must be a bank statement, a screenshot of your online banking environment, a letter from the bank or a cheque and must contain \
+        the logo of the bank or it's name in a unique font, the bank account details, the name of the account holder.\
+        Allowed formats: jpg, pdf, png. Maximum allowed size: 10MB.")
+    bank_statement_filename = fields.Char()
+
+    # KYC
+    adyen_kyc_ids = fields.One2many('adyen.kyc', 'bank_account_id')
+    kyc_status = fields.Selection(ADYEN_KYC_STATUS, compute='_compute_kyc_status', readonly=True)
+    kyc_status_message = fields.Char(compute='_compute_kyc_status', readonly=True)
+
+    @api.depends_context('lang')
+    @api.depends('adyen_kyc_ids')
+    def _compute_kyc_status(self):
+        self.kyc_status_message = False
+        self.kyc_status = False
+        for bank_account in self.filtered('adyen_kyc_ids'):
+            kyc = bank_account.adyen_kyc_ids._sort_by_status()
+            bank_account.kyc_status = kyc[0].status
+
+    @api.model
+    def create(self, values):
+        adyen_account_id = self.env['adyen.account'].browse(values.get('adyen_account_id'))
+        response = adyen_account_id._adyen_rpc('v1/update_account_holder', self._format_data(values))
```

Comment:
Is it on purpose to call this before the super? If it fails (missing field, wrong access rights, constraint...) it will still make the rpc.
Same comment for write and unlink below

Reply:
It's done on purpose to only create the record if it successfully goes through the API (to prevent having "ghost" records), I've added explicit checks now where needed :-)

Reply:
I tried to reproduce the case that was causing issues but somehow couldn't :facepalm: 

I've adapted `adyen.account`, `adyen.bank.account` and `adyen.shareholder` to call super() before the API call for `create` and `write`.